File: backend/app/schemas/payment.py
"""
Payment schemas for sandbox QR flow.
"""
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel


class PaymentSessionCreate(BaseModel):
    """Create payment session request schema."""

    event_id: str
    category: str
    amount: int


class PaymentSessionResponse(BaseModel):
    """Payment session response schema."""

    id: str
    event_id: str
    category: str
    amount: int
    status: str
    created_at: datetime
    expires_at: Optional[datetime] = None


class PaymentConfirmRequest(BaseModel):
    """Confirm or cancel a payment session."""

    session_id: str
    action: Literal["confirm", "cancel"] = "confirm"

File: lib/api/payment.ts
import apiClient from './client'

export interface PaymentSession {
  id: string
  event_id: string
  category: string
  amount: number
  status: string
  created_at: string
  expires_at?: string | null
}

export interface CreatePaymentSessionRequest {
  event_id: string
  category: string
  amount: number
}

export async function createPaymentSession(
  data: CreatePaymentSessionRequest
): Promise<PaymentSession> {
  const response = await apiClient.post('/payment/session', data)
  return response.data
}

export async function getPaymentSession(sessionId: string): Promise<PaymentSession> {
  const response = await apiClient.get(`/payment/session/${sessionId}`)
  return response.data
}

export async function confirmPaymentSession(
  sessionId: string,
  action: 'confirm' | 'cancel' = 'confirm'
): Promise<{ status: string }> {
  const response = await apiClient.post('/payment/confirm', {
    session_id: sessionId,
    action,
  })
  return response.data
}


